File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public static void dfs(TreeNode root, List<Integer> res, int size) {
        if (root == null) {
            return;
        }

        if (res.size() <= size) {
            res.add(root.val);
        }
        dfs(root.right, res, size + 1);
        dfs(root.left, res, size + 1);
    }
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        dfs(root, res, 0);
        return res;
    }
}

File: Data Structures & Algorithms/perfect-squares/submission-0.cpp
class Solution {
public:
    unordered_map<int,int> dp;
    int dfs(int t) {
        if (t == 0) return 0;
        if (dp.count(t)) return dp[t];

        int res = t;
        for(int i = 1; i * i <= t; i++){
            res = min(res,1+dfs(t - i*i));
        }
        return dp[t] = res;
    }
    int numSquares(int n) {
        return dfs(n);
    }
};